python/cudf/cudf/core/index.py

```diff
@@ -1078,8 +1078,10 @@ def as_index(arbitrary, **kwargs):
         return RangeIndex(start=arbitrary.start, stop=arbitrary.stop, **kwargs)
     elif isinstance(arbitrary, pd.MultiIndex):
         return cudf.MultiIndex.from_pandas(arbitrary)
-    else:
-        return as_index(column.as_column(arbitrary), **kwargs)
+    elif isinstance(arbitrary, range):
+        if arbitrary.step == 1:
```

Comment:
Should we error if step isn't 1 as opposed to falling back to a non-RangeIndex?

Reply:
I couldn't decide. We should totally just support `step` in our `RangeIndex` and until then I think let's do this.